de.fhdo.lemma.servicedsl.extractor/src/de/fhdo/lemma/servicedsl/extractor/ServiceDslExtractor.xtend

```diff
@@ -0,0 +1,272 @@
+import de.fhdo.lemma.technology.CommunicationType
+import de.fhdo.lemma.technology.ExchangePattern
+import de.fhdo.lemma.technology.Protocol
+import de.fhdo.lemma.service.ImportedServiceAspect
+
+
+/**
+ * Model-to-text extractor for the Service DSL.
+ *
+ * @author <a href="mailto:jonas.sorgalla@fh-dortmund.de">Jonas Sorgalla</a>
+ */
+class ServiceDslExtractor {
+    static val IMPORTED_TECH_ALIASES = <String>newArrayList()
+    static val ID_PATTERN = "(\\^?)([a-zA-Z_])\\w*"
+    static val QUALIFIED_NAME_PATTERN = '''«ID_PATTERN»(\.«ID_PATTERN»)*'''
+    static val QUALIFIED_NAME_WITH_AT_LEAST_ONE_LEVEL_PATTERN = '''«ID_PATTERN»\.''' +
+        QUALIFIED_NAME_PATTERN
+
+    /**
+     * Extract ServiceModel
+     */
+    def String extractToString(ServiceModel serviceModel) {
+        val imports = <String>newArrayList
+        serviceModel.imports.forEach[imports.add(it.generate)]
+        val importStatements = if (!imports.empty)
+                String.join("\n", imports) + "\n\n"
+            else
+                ""
+
+        val microservices = String.join("\n\n", serviceModel.microservices.map[generate])
+
+        return '''«importStatements»«microservices»'''
+    }
+
+    /**
+     * Extract Imports
+     */
+    private def generate(Import ^import) {
+        val importTypeKeyword = switch(^import.importType) {
+            case DATATYPES: "datatypes"
+            case MICROSERVICES: "microservices"
+            case TECHNOLOGY: "technology"
+            default: null
+        }
+        // If it is a technology add it to the names list for later use in the generation
+        if (importTypeKeyword == "technology") 
+        	IMPORTED_TECH_ALIASES.add(^import.name)
+
+        return '''import «importTypeKeyword» from "«^import.importURI»" as «^import.name»'''
+    }
+
+    /**
+     * Extract Microservice
+     */
+    private def generate(Microservice service) { 
+        val preamble = '''«service.visibility.generate» «service.type.generate»'''
+        return '''
+        «service.generateTechAnnotation»
+        «preamble» microservice «service.lemmaName» {
+            «IF service.interfaces.exists[!operations.empty]»
+                «FOR iface : service.interfaces»
+                    «iface.generate»
+                «ENDFOR»
+            «ELSE»
+                [DEFINE_OPERATIONS]
+            «ENDIF»
+        }'''
+    }
+
+    /**
+     * Extracts the name of a Microservice
+     */
+    private def lemmaName(Microservice service) {
+        return if (service.name.matches(QUALIFIED_NAME_WITH_AT_LEAST_ONE_LEVEL_PATTERN))
+                service.name
+            else
+                '''ADD_QUALIFYING_PART.«service.name»'''
+    }
+
+    /**
+     * Extract Visibility of a Microservice
+     */
+    private def generate(Visibility visibility) {
+        return switch(visibility) {
+            case INTERNAL: 'internal'
+            case PUBLIC: 'public'
+            default: 'architecture'
```

Comment:
Would be more consistent with the rest of the extractor to throw an `IllegalArgumentException` as `default` case.